Added display of immediate operands to disassembly

weijuwang opened the pull request:
This is for issue #5. I replaced all the placeholders in the disassembly table with either `%hhu` for 8-bit immediates or `%hu` for 16-bit immediates. These are formats that `printf` understands. When disassembling an instruction, the code then looks for these substrings in the instruction's assembly representation and, if needed, passes the value of the immediate from memory as an argument to `printf` so that it can be displayed.

weijuwang committed:
Added display of immediate values to disassembly

The debug output's disassembly now displays immediate values passed to instructions, when applicable.

weijuwang commented on i8080.h:
```diff
@@ -4,6 +4,7 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <stdbool.h>
+#include <string.h>
```
`strstr()`

superzazu commented:
Hello, thank you for the contribution!

Can you use `%02X` and `%04X` instead of `%hu` and `% hhu`? It would be more consistent with the rest of the project, and more useful (I think?) to display those values as hexadecimal.

weijuwang committed:
Changed formats of immediates to hexadecimal

%hu -> %02X, %hhu -> %04X
Since immediates are being displayed in hex, an H is also added to the end of the value.

weijuwang commented:
Should be fixed now.

weijuwang commented:
On a somewhat related note: the disassembly could be made more readable if some more spaces were added, e.g.
```
... cpo 06A0H
... cm  06A0H
... cnz 06A0H
... cpi 00H
```
as opposed to
```
... cpo 06A0H
... cm 06A0H
... cnz 06A0H
... cpi 00H
```
I could make a separate PR if needed, since it's not strictly related to immediate operands.
